libraries/botbuilder-core/src/cloudAdapterBase.ts

```diff
@@ -176,6 +179,82 @@ export abstract class CloudAdapterBase extends BotAdapter {
         return this.processProactive(claimsIdentity, ActivityEx.getContinuationActivity(reference), audience, logic);
     }
 
+    /**
+     * @inheritdoc
+     */
+    async createConversationAsync(
+        botAppId: string,
+        channelId: string,
+        serviceUrl: string,
+        audience: string,
+        conversationParameters: ConversationParameters,
+        logic: (context: TurnContext) => Promise<void>
+    ): Promise<void> {
+        // Create a ClaimsIdentity, to create the connector and for adding to the turn context.
+        const claimsIdentity = this.createClaimsIdentity(botAppId);
+        claimsIdentity.claims.push({ type: AuthenticationConstants.ServiceUrlClaim, value: serviceUrl });
+
+        // Create the connector factory.
+        const connectorFactory = this.botFrameworkAuthentication.createConnectorFactory(claimsIdentity);
+
+        // Create the connector client to use for outbound requests.
+        const connectorClient = await connectorFactory.create(serviceUrl, audience);
+
+        // Make the actual create conversation call using the connector.
+        const createConversationResult = await connectorClient.conversations.createConversation(conversationParameters);
+
+        // Create the create activity to communicate the results to the application.
+        const createActivity = this.createCreateActivity(
+            createConversationResult.id,
+            channelId,
+            serviceUrl,
+            conversationParameters
+        );
+
+        // Create a UserTokenClient instance for the application to use. (For example, in the OAuthPrompt.)
+        const userTokenClient = await this.botFrameworkAuthentication.createUserTokenClient(claimsIdentity);
+
+        // Create a turn context and run the pipeline.
+        const context = this.createTurnContext(
+            createActivity,
+            claimsIdentity,
```

Comment:
I would expect that at this point, the `claimsIdentity` would have an audienceClaim with the value of `audience`, not the `botAppId`:

https://github.com/microsoft/botbuilder-js/blob/0db823143a83267c60d42a7daef2cb04cc72ee4a/libraries/botbuilder-core/src/cloudAdapterBase.ts#L307-L315

Is there a reason this isn't the case?

What will happen is that the `ClaimsIdentity` is essentially Emulator-esque as the appId and audience claim are the same value (the `botAppId`), whereas the created `ConnectorClient` is properly addressed to the passed in `audience`. This `audience` will can be a channel, another bot, or something else entirely.

Reply:
Not sure, I followed John's lead in porting this. As far as I can tell, this is equivalent [to .NET](https://github.com/microsoft/botbuilder-dotnet/blob/54c75546821c3e4b90d7b24a3f98da549761e753/libraries/Microsoft.Bot.Builder/CloudAdapterBase.cs#L321). Perhaps we should file an issue on .NET?